Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -191,6 +200,8 @@ class CustomerIOImplementation: CustomerIOInstance {
         name: String,
         data: RequestBody?
     ) {
+        CIODataPipeline.analytics.track(name: name, properties: jsonAdapter.toJson(data))
```

Comment:
There are multiple `track` functions in this file with different function parameters. All of the `track` variations call 1 common function, `trackEvent`. 

Shouldn't this line go into `trackEvent`? Otherwise, if a customer called, for example, `track(name: String, data: [String: Any])` function, then that request will not get forwarded to the CDP module. 

If the function `trackEvent` is confusing, feel free to make refactors you need. The goal of the `trackEvent` function is to be 1 function where all event and screen tracking logic exists. You can rename to `commonTrack` or something if that makes more sense to you.

Reply:
```
public func track<RequestBody: Encodable>(
        name: String,
        data: RequestBody?
    ) {
        CIODataPipeline.analytics.track(name: name, properties: jsonAdapter.toJson(data))

        _ = trackEvent(type: .event, name: name, data: data)
    }

    public func track(name: String, data: [String: Any]) {
        track(name: name, data: StringAnyEncodable(logger: logger, data))
    }
```
The other track method seems to be calling the track method i added the CDP redirection in. I didn't add the code in the `trackEvent` because it contains logic for both screen and track, while we don't have to combine it as Segment had separate methods for both. 

And i don't see us keeping  method `track<RequestBody: Encodable>`  as it was supposed to be used internally and was helpful when we were using our BQ, if we are just redirect these calls, we don;t have use of it. Given track get removed from our SDK.

Reply:
I am glad to hear that the CDP module call will be called by all versions of `track()`. 

I do still prefer to have all logic for tracking events inside of 1 function. For code readability and maintainability, I think it would be better to have all logic encapsulated together instead of having some code in a common function and some code not.

Reply:
I understand it would be easier, i can add a `todo` for that. My hesitation is, i don't wan't to unncessarily rely on the data classes we have created for our tracking and BQ because they might get removed. Simiarly the `trackEvent` method might not even exist and neither would the `track<RequestBody: Encodable>(name: String,data: RequestBody?`. 

The only method that would remain would be 

```
track(name: String, data: [String: Any])

and it would call 
        CIODataPipeline.analytics.track(name: name, properties: data)

```

Reply:
Ah, ok. 

I would be OK with adding a todo and deciding after the fate of tracking code is determined. Good idea.